### PR TITLE
HIP Fix Number of threads in TeamPolicy

### PR DESCRIPTION
**Pull request by PaulMullowney**
Set number of threads to 128 rather than use Kokkos::AUTO

### CODE REVIEW

**Review by jrood-nrel — APPROVED** [review]

**Comment by PaulMullowney** [comment]
Actually. Don't merge this yet. I found another example that I want to fix.

File: src/Simulation.C [hunk 1]
```diff
@@ -56,8 +56,8 @@ Simulation::Simulation(const YAML::Node& root_node)
 #endif
 
 #ifdef KOKKOS_ENABLE_HIP
-  // hipDeviceGetLimit(&default_stack_size, cudaLimitStackSize);
-  // hipDeviceSetLimit(cudaLimitStackSize, nalu_stack_size);
+  hipDeviceGetLimit(&default_stack_size, hipLimitMallocHeapSize);
+  hipDeviceSetLimit(hipLimitMallocHeapSize, nalu_stack_size);
```
Comment (alanw0) [inline src/Simulation.C:60]: Excellent, I had thought that we believed these functions were not yet available. Are they newly available in 5.2 or later?

**Review by alanw0 — APPROVED** [review]

File: src/Simulation.C (hunk 1 above)
Comment (PaulMullowney) [reply src/Simulation.C:60]: We need 5.3 or later for these

Comment (PaulMullowney) [reply src/Simulation.C:60]: Given that 5.4 seems to be the default in spack-manager, I think we're ok

Comment (psakievich) [reply src/Simulation.C:55]: We can put a hard constraint on nalu-wind in spack that makes it so it won't build with anything less than 5.3 when needed.

File: include/ngp_utils/NgpLoopUtils.h [hunk 2]
```diff
@@ -42,7 +42,11 @@ inline TeamPolicy
 ngp_mesh_team_policy(
   const size_t sz, const size_t bytes_per_team, const size_t bytes_per_thread)
 {
+#if defined(KOKKOS_ENABLE_HIP)
+  TeamPolicy policy(sz, 128);
```
Comment (psakievich) [inline include/ngp_utils/NgpLoopUtils.h:46]: Can we make this number a macro so it only needs to be changed in one place going forward?

**Review by psakievich — CHANGES_REQUESTED** [review]

File: include/ngp_utils/NgpLoopUtils.h (hunk 2 above)
Comment (PaulMullowney) [reply include/ngp_utils/NgpLoopUtils.h:46]: > Can we make this number a macro so it only needs to be changed in one place going forward?

Good idea

**Comment by PaulMullowney** [comment]
Ready to merge. We're at 80% of the unit tests passing which is pretty good.

**Review by psakievich — APPROVED** [review]

**Comment by psakievich** [comment]
@ddement let's try to avoid merging PR's that are failing the style check.

**Pull request merged**